flink-ml-lib/src/test/java/org/apache/flink/ml/classification/NaiveBayesTest.java

```diff
@@ -0,0 +1,349 @@
+        NaiveBayesModel modelB = new NaiveBayesModel().setModelData(modelData);
+        ReadWriteUtils.updateExistingParams(modelB, modelA.getParamMap());
+
+        checkResult(modelB);
+    }
+
+    private void runAndCheck() throws Exception {
+        NaiveBayesModel model = getModel();
+        checkResult(model);
+    }
+
+    private NaiveBayesModel getModel() throws Exception {
+        Table trainTable =
+                tEnv.fromDataStream(
+                                env.fromElements(trainData)
+                                        .assignTimestampsAndWatermarks(
+                                                WatermarkStrategy.noWatermarks()),
+                                schema)
+                        .as("weight", "features", "label");
+
+        NaiveBayes estimator =
+                new NaiveBayes()
+                        .setSmoothing(smoothing)
+                        .setFeaturesCol(featuresCol)
+                        .setLabelCol(labelCol)
+                        .setPredictionCol(predictCol)
+                        .setModelType(modelType);
+
+        if (isSaveLoad) {
+            String tempDir = Files.createTempDirectory("").toString();
+            estimator.save(tempDir);
+            env.execute();
+
+            estimator = NaiveBayes.load(tempDir);
+        }
+
+        NaiveBayesModel model = estimator.fit(trainTable);
+
+        if (isSaveLoad) {
+            String tempDir = Files.createTempDirectory("").toString();
+            model.save(tempDir);
+            env.execute();
+
+            model = NaiveBayesModel.load(tempDir);
+        }
+
+        return model;
+    }
+
+    private void checkResult(NaiveBayesModel model) {
+        Table predictTable =
+                tEnv.fromDataStream(
+                                env.fromElements(predictData)
+                                        .assignTimestampsAndWatermarks(
+                                                WatermarkStrategy.noWatermarks()),
+                                schema)
+                        .as("weight", "features", "label");
+
+        Table output =
+                model.transform(predictTable)[0].select(
+                        $("weight"), $("features"), $("label"), $("predict"));
+
+        Object[] actualObjects = IteratorUtils.toArray(output.execute().collect());
+        Row[] actual = new Row[actualObjects.length];
+        for (int i = 0; i < actualObjects.length; i++) {
+            actual[i] = (Row) actualObjects[i];
+        }
+
+        assertEquals(errorMessage, getFrequencyMap(expectedOutput), getFrequencyMap(actual));
+    }
+
+    private static Map<Object, Integer> getFrequencyMap(Row[] rows) {
+        Map<Object, Integer> map = new HashMap<>();
+        for (Row row : rows) {
+            List<Object> list = toList(row);
+            map.put(list, map.getOrDefault(list, 0) + 1);
+        }
+        return map;
+    }
+
+    private static List<Object> toList(Row row) {
```

Comment:
Would it be better to add a `List<Object> getFieldValues(Row row)` in `TableUtils`?

Reply:
classes like this has been replaced by existing tools like `CollectionUtils` in `org.apache.commons`.